### PR TITLE
(compilers): add version and pname

### PR DESCRIPTION
**Pull request by Ericson2314**
###### Motivation for this change

Re-sending #67284 to staging, where it was meant to go

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/build-support/cc-wrapper/default.nix [hunk 1]
```diff
@@ -96,6 +96,7 @@ stdenv.mkDerivation {
   name = targetPrefix
     + (if name != "" then name else stdenv.lib.removePrefix targetPrefix "${ccName}-wrapper")
     + (stdenv.lib.optionalString (cc != null && ccVersion != "") "-${ccVersion}");
+  version = ccVersion;
```
Comment (veprbl) [inline pkgs/build-support/cc-wrapper/default.nix]: My https://github.com/NixOS/nixpkgs/pull/67284#discussion_r349885907 was never addressed. If we are really going to break the case when `cc == null`, we might as well change this to provide pname.

Comment (veprbl) [reply pkgs/build-support/cc-wrapper/default.nix]: Ah never mind, didn't notice 591034b45a83f28734afdc5ed787141f8494a6cc . This is not very pretty.

Comment (Ericson2314) [reply pkgs/build-support/cc-wrapper/default.nix]: Oh, sorry I missed that part. I agree.

**Comment by Ericson2314** [comment]
Closing cause I can't push to another branch if I opened the PR.

**Pull request closed without merging**